src/compiler/binder.ts

```diff
@@ -2608,10 +2619,15 @@ namespace ts {
 
         // If a FunctionExpression contains an asterisk token, or its subtree has marked the container
         // as needing to capture the lexical this, then this node is ES6 syntax.
-        if (asteriskToken || (subtreeFlags & TransformFlags.ES6FunctionSyntaxMask)) {
+        if (subtreeFlags & TransformFlags.ES6FunctionSyntaxMask) {
             transformFlags |= TransformFlags.AssertES6;
         }
 
+        // Currently, we only support generators that were originally async function bodies.
+        if (asteriskToken && node.emitFlags & NodeEmitFlags.AsyncFunctionBody) {

```

Comment:
`TransformFlags` aggregation happens both during the bind phase and during each successive transformation phase (though usually it has less work to do as it can reuse the flags from unchanged subtrees).

In this case, we only add the flag if the generator was created as a result of an async function transformation.

We do not down-level generators that are not the result of an async function transformation.